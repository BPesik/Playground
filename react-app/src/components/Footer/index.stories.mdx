import Footer from "./index";
import { Props } from "@storybook/addon-docs/blocks";

<Meta title="Footer" 
    component={Footer} 
    argTypes={{}}
/>

export const Template = (args) => <Footer {...args} />;

<Canvas>
    <Story name="Primary" args={{}}>
        {Template.bind({})}
    </Story>
</Canvas> 

# Footer

The footer component requires no properties
