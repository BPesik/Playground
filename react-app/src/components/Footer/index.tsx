import React from "react";
import styled from "@emotion/styled";

const StyledFooter = styled.footer`
    background-color: grey;
    bottom: 0;
    height: 40px;
    left: 0;
    position: fixed;
    text-align: center;
    width: 100%;
`;

const Link = styled.a`
    color: white;
    font-family: Arial;
    position: absolute;
    top: 25%;
`;

function Footer() {
    return (
        <StyledFooter>  
            <Link href=""
                target="_blank"
                rel="noopener noreferrer">
                    Learn React
            </Link>
        </StyledFooter>
    )
}

export default Footer;
