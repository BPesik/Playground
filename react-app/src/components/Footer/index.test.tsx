import React from "react";
import { render } from "enzyme";
import Footer from "./index"

describe("The Footer component", () => {
    const wrapper = render(<Footer />)
    
    it("should match snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("should render the correct link", () => {
        expect(wrapper[0].firstChild.name).toBe("a");
        expect(wrapper[0].firstChild.firstChild.data).toBe("Learn React");
    });
});
