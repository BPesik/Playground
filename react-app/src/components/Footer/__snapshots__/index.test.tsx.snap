// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Footer component should match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-1xe7u6f",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-1mqj3gt",
          "href": "",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "children": Array [
          Object {
            "data": "Learn React",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "href": undefined,
          "rel": undefined,
          "target": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "href": undefined,
          "rel": undefined,
          "target": undefined,
        },
      },
    ],
    "name": "footer",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
